docs/docs/50-user-guide/20-how-to-guides/30-working-with-warehouses.md

```diff
@@ -344,6 +352,114 @@ strategies are:
           allowTags: ^nightly
     ```
 
+#### Expression Filtering
+
+Git repository subscriptions support advanced filtering using expressions. These
+expressions allow you to filter commits and tags based on their metadata using
+[expr-lang](https://expr-lang.org) syntax.
+
+:::info
+The expressions must evaluate to a boolean value (`true` or `false`). If an
+expression evaluates to a non-boolean value, an attempt will be made to
+convert it to a boolean (e.g., `0` to `false`, `1` to `true`).
+:::
+
+:::warning
+Invalid expressions will cause the subscription to fail. Always test your
+expressions to ensure they evaluate correctly with your repository's data.
+:::
+
+:::tip
+You can test your expressions using the
+[expr-lang playground](https://expr-lang.org/playground).
+
+The playground allows you to evaluate expressions against sample data and
+see the results in real-time. This is especially useful for debugging and
+validating your expressions before applying them to your `Warehouse` resources.
+:::
+
+##### Tag Expression Filtering
```

Comment:
Can you explain why a `tagExpressionFilter` is necessary in addition to `commitExpressionFilter`? I thought everything is a commit at the end of the day and we would just need `commitExpressionFilter`. I noticed the fields that are available in both of them (`tag`, `commitID`, `author`, etc...) are identical. Would they ever be different?